### DISCUSSION TITLE
How to send file via sftp in javascript

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6293] [Q&A]
I'm trying to create a code template that will sftp a file. I found some code using a sftpClient but Mirth doesn't recognize that. I'm thinking I need to import a java library but not sure which one.

I was using the below code:

```
var sftpClient = new SftpClient();
sftpClient.connect(sftpUrl,sftpUser,sftpPwd,port);
```

Would appreciate any help or a better way to do it.

**Comment by pacmano1** [comment]
https://github.com/pacmano1/Mirth-Snippets/blob/main/sftp_get_to_attachment_newJsch.js

**Reply by pacmano1** [reply]
just something you can work with there.

**Reply by rutledj** [reply]
Thanks. That was helpful. But I don't think I can make it work as a code template as I will be passing in an arraylist of actual channels from the postgres channels table. Looks like I need to create an actual file but I don't have a local folder available to create a file. Not sure if one can be created just in memory.

**Reply by pacmano1** [reply]
I don't understand your use case.
